tests/test_data.py

```diff
@@ -4,213 +4,155 @@
+        )
+
+        with Source(datahub=Datahub.Scihub) as src:

```

Comment:
I have no experience with mocking request, but where do you use variable m and aren't you still querying the external hub with query metadata? Please ignore if you think this is correct.

Reply:
See https://requests-mock.readthedocs.io/en/latest/mocker.html#decorator